contracts/src/gateway/router/XnetMessagingFacet.sol

```diff
@@ -29,4 +29,28 @@ contract XnetMessagingFacet is GatewayActorModifiers {
     function applyCrossMessages(IpcEnvelope[] calldata crossMsgs) external systemActorOnly {
         LibGateway.applyMessages(s.networkName.getParentSubnet(), crossMsgs);
     }
+
+    /// @notice Applies top-down cross-net message locally one by one.
+    ///         This is invoked by IPC nodes when drawing messages from their parent
+    ///         subnet for local execution. That's why the sender is restricted to the system sender,
+    ///         because this method is implicitly invoked by the node during block production.
+    /// @dev It requires the caller to be the system actor.
+    /// @param crossMsg The cross-network message to be applied.
+    function applyMsgWithRet(
+        IpcEnvelope calldata crossMsg
+    ) external systemActorOnly returns (bool hasReturn, ApplyMsgRet memory) {
```

Comment:
Just so I understand: `hasReturn` indicates that the second member is not a default value?